pipeline/store/accessor.go

```diff
@@ -0,0 +1,177 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/streamingfast/dstore"
+	"github.com/streamingfast/substreams/block"
+	pbsubstreams "github.com/streamingfast/substreams/pb/sf/substreams/v1"
+	"github.com/streamingfast/substreams/state"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var ErrStoreNotFound = errors.New("store not found")
+
+type Accessor struct {
+	baseStore           dstore.Store
+	storeMap            map[string]*state.Store
+	SaveInterval        uint64
+	maxBoundaryBlockNum uint64
+	nextSaveBoundary    uint64
+	logger              *zap.Logger
+}
+
+func NewAccessor(saveInterval, minBoundaryBlock, maxBoundaryBlockNum uint64, baseStore dstore.Store, logger *zap.Logger) *Accessor {
```

Comment:
The accessor now has too many responsibilies.

We had discussed a `StoreMap` which would simply pass the map around, and something like a `store.MapGetter` or something, that would allow downstream to simply get the stores and not assign them.

Accessor has grown to be much more.

I remember even thinking the concept of a Store holding its InitialBlock was already too much. A Store ought not to know how it's being used. Perhaps the concept of a `PartialStore` could hold such a number.. with the `Roll()` method.. In normal linear processing, that number is not useful and is just confusing. I remember myself being confused by it.

So a new idea would be to embed the Store in a PartialStore, and use PartialStore where we expect a partial store (in the orchestrated stuff), but not everywhere. The Squasher for example, works with both types, and having clarity as to which is which would be really useful.

Reply:
Or a `BoundedStore` ? which knows its boundaries?

There's a StoreReadWriter / StoreReader and StoreWriter in here.. that knows how to Load and Save itself from storage.

There's signals the StoreMap could dispatch to the Store or PartialStore depending.. so would relieve the Pipeline from knowing what will actually happen with the stores, when certain signals arrive (like BlockComplete or something).

Maybe we also think here that we only trigger on Final signals, because we don't want to save stores in the reversible segments. Is that propery handled?

Perhaps StoreMap has the same things as the ForkHandler, something like `Init`, `NewFinalBlock`, `UndoBlock` or whatnot? Same as the ExecOutput has? That would be nice, if every component knew what to do with the reorg signals.

Reply:
I need more context here. What is the difference between 

`PartialStore`
`Store`
`StoreMap`